jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamCommon.kt

```diff
@@ -33,6 +34,27 @@ class SamCommon {
         // Exclusive
         val expectedSamMaxVersion = SemVer("0.16.0", 0, 16, 0)
 
+        private val versionCache = object : FileInfoCache<SemVer>() {
```

Comment:
Assume this looks at the file system update time for the file before expiring the cache entry?

Reply:
yes - just looked at impl sorry, I thought this might be one of the JB pre-baked utils.